csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/Runtime.cs

```diff
@@ -75,21 +78,38 @@ private static IEnumerable<string> DesktopRuntimes
                         .OrderByDescending(Path.GetFileName);
                 }
 
-                var monoPath = FileUtils.FindProgramOnPath(Win32.IsWindows() ? "mono.exe" : "mono");
-                var monoDirs = monoPath is not null
-                    ? new[] { Path.GetFullPath(Path.Combine(monoPath, "..", "lib", "mono")), monoPath }
-                    : new[] { "/usr/lib/mono", "/usr/local/mono", "/usr/local/bin/mono", @"C:\Program Files\Mono\lib\mono" };
+                string? monoDir = null;
 
-                var dir = monoDirs.FirstOrDefault(Directory.Exists);
+                var monoPathEnv = Environment.GetEnvironmentVariable(EnvironmentVariableNames.MonoPath);
```

Comment:
Should this path be mono specific? We could add instead `CODEQL_EXTRACTOR_CSHARP_BUILDLESS_NET_FRAMEWORK_PATH`, and then we could also cover the above Windows branch too.

Reply:
Yes, that is a good idea!
We should we able to "override" the entire logic in `DesktopRuntimes` by setting the environment variable.